runtime/vm/LayoutFFITypeHelpers.cpp

```diff
@@ -100,31 +101,94 @@ LayoutFFITypeHelpers::getArrayFFIType(char **layout, UDATA nElements) {
 	return typeFFI;
 }
 
-ffi_type*
+ffi_type *
 LayoutFFITypeHelpers::getStructFFIType(char **layout)
 {
+	char *currentLayout = *layout;
+	J9LayoutStrFFITypeEntry *layoutStrFFITypeEntry = NULL;
+	J9LayoutStrFFITypeEntry *resultEntry = NULL;
+	ffi_type **structElements = NULL;
 	ffi_type *typeFFI = NULL;
+	PORT_ACCESS_FROM_JAVAVM(_vm);
+
+	if (NULL == _vm->layoutStrFFITypeTable) {
+		_vm->layoutStrFFITypeTable = createLayoutStrFFITypeTable(_vm);
+		if (NULL == _vm->layoutStrFFITypeTable) {
+			goto freeAllMemoryThenExit;
+		}
+	}
+
+	layoutStrFFITypeEntry = (J9LayoutStrFFITypeEntry *)j9mem_allocate_memory(sizeof(J9LayoutStrFFITypeEntry), J9MEM_CATEGORY_VM_FFI);
+	if (NULL == layoutStrFFITypeEntry) {
+		goto freeAllMemoryThenExit;
+	}
+
+	/* Search the hashtable for the ffi_type if the same strcut layout string exists. */
```

Comment:
Please correct the spelling of "struct". In the header file, "a strcut arrays" should be "a struct array".

Reply:
> Please fix the typos on lines 219 and 279: "bytes is required" should be "bytes are required".

Fixed.

Reply:
> Please correct the spelling of "struct". In the header file, "a strcut arrays" should be "a struct array".

Fixed.